modules/SRanipal_module/src/GazeRetargeting.cpp

```diff
@@ -0,0 +1,695 @@
+    }
+
+    double ipd = 0.0;
+
+    if (!getValueFromRPC("getIPD", ipd))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to retrieve the IPD from the VR device.";
+        return false;
+    }
+
+    m_leftEye.eyePosition.zero();
+    m_leftEye.eyePosition(0) = -ipd/2.0;
+    m_rightEye.eyePosition.zero();
+    m_rightEye.eyePosition(0) = ipd/2.0;
+
+    double eyesZPosition = -1.0;
+
+    if (!getValueFromRPC("getEyesZPosition", eyesZPosition))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to retrieve the eyes Z position from the VR device.";
+        return false;
+    }
+
+    m_leftEye.imageRelativePosition.zero();
+    m_leftEye.imageRelativePosition(2) = eyesZPosition;
+    m_rightEye.imageRelativePosition.zero();
+    m_rightEye.imageRelativePosition(2) = eyesZPosition;
+
+    std::string leftEyePortName;
+
+    if (!getValueFromRPC("getLeftImageControlPortName", leftEyePortName))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to retrieve the left image control port name from the VR device.";
+        return false;
+    }
+
+    std::string leftEyeOutputPortName = "/" + m_name + "/leftEye/control:o";
+
+    if (!m_leftEye.controlPort.open(leftEyeOutputPortName))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to open the port" << leftEyeOutputPortName;
+        return false;
+    }
+
+    if (!yarp::os::Network::connect(leftEyeOutputPortName, leftEyePortName))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to connect the port" << leftEyeOutputPortName << "to" << leftEyePortName;
+        return false;
+    }
+
+    std::string rightEyePortName;
+
+
+    if (!getValueFromRPC("getRightImageControlPortName", rightEyePortName))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to retrieve the right image control port name from the VR device.";
+        return false;
+    }
+
+    std::string rightEyeOutputPortName = "/" + m_name + "/rightEye/control:o";
+
+    if (!m_rightEye.controlPort.open(rightEyeOutputPortName))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to open the port" << rightEyeOutputPortName;
+        return false;
+    }
+
+    if (!yarp::os::Network::connect(rightEyeOutputPortName, rightEyePortName))
+    {
+        yError() << "[GazeRetargeting::VRInterface::isActive] Failed to connect the port" << rightEyeOutputPortName << "to" << rightEyePortName;
+        return false;
+    }
+
+    setVRImagesPose(0.0, 0.0, 0.0);
+
+    m_isActive = true;
+
+    yInfo() << "[GazeRetargeting::VRInterface::isActive] Gaze retargeting ready!";
+
+    return true;;
+}
+
+void GazeRetargeting::VRInterface::close()
+{
+    m_VRDeviceRPCOutputPort.close();
+    m_leftEye.close();
+    m_rightEye.close();
+}
+
+void GazeRetargeting::VRInterface::EyeControl::sendAngles()
+{
+    yarp::sig::Vector& output = controlPort.prepare();
```

Comment:
```suggestion
    yarp::sig::Vector& output = imageControlPort.prepare();
```

Reply:
Good point. Done in https://github.com/robotology/walking-teleoperation/pull/82/commits/2268993a86729c662064994ac874fbda3515e35b